### PR TITLE
11.0 mig base user role

### PR DESCRIPTION
**Pull request by novawish**

### CODE REVIEW

**Review by hieulucky111 — APPROVED** [review]

File: base_user_role/data/ir_cron.xml [hunk 1]
```diff
@@ -3,16 +3,17 @@
...
+        <field name='interval_number'>3</field>
+        <field name='interval_type'>hours</field>
+        <field name="numbercall">-1</field>
+        <field name="active">True</field>
+        <field name="doall" eval="False"/>
+        <field name="model_id" ref="base_user_role.model_res_users_role"/>
+        <field name="state">code</field>
+        <field name="code">model.cron_update_users()</field>
+        <field name="args">()</field>
+    </record>
```
Comment (sebalix) [inline base_user_role/data/ir_cron.xml]: Is it worth it to reindent all the XML (I take the OCA template module as the reference: https://github.com/OCA/maintainer-tools/blob/master/template/module/data/model_name_data.xml)? It makes big diff for no value, and will make harder to backport&merge code from older branches.

**Review by sebalix — CHANGES_REQUESTED** [review]
Thank you! I would prefer to not reindent XML data, see my comment below. What do you think?

File: base_user_role/data/ir_cron.xml (hunk 1 above)
Comment (novawish) [reply base_user_role/data/ir_cron.xml]: I think you're right, thanks for you comment :+1:

File: base_user_role/views/role.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2014 ABF OSIELL <http://osiell.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+<record id="view_res_users_role_form" model="ir.ui.view">
```
Comment (elicoidal) [inline base_user_role/views/role.xml:6]: Shouldnot this level be indented?

**Review by elicoidal — APPROVED** [review]
small nitpick.
LGTM (no test)

File: base_user_role/views/role.xml (hunk 2 above)
Comment (novawish) [reply base_user_role/views/role.xml:6]: @elicoidal  See [above comment](https://github.com/OCA/server-backend/pull/1#discussion_r151944395) of @sebalix

Comment (elicoidal) [reply base_user_role/views/role.xml:6]: OK so we are in line :smile:

File: base_user_role/data/ir_cron.xml (hunk 1 above)
Comment (sebalix) [reply base_user_role/data/ir_cron.xml]: @novawish it seems that the XML ./data/ files should be fixed as the view ones

Comment (novawish) [reply base_user_role/data/ir_cron.xml]: Done! I also squashed all commits

**Review by sebalix — APPROVED** [review]
Tested locally :+1: thank you

**Review by sebalix — CHANGES_REQUESTED** [review]
Oups sorry, I think we should integrate this fix from the 8.0 branch https://github.com/OCA/server-tools/commit/c83d4b5724ad113625db1b1a537fa35f4e686fda#diff-34ee657308a4ece3930a50c4dfe4c56f (a cherry-pick should be enough)

**Comment by sebalix** [comment]
The new code raise this error:

```python
Traceback (most recent call last):
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/odoo/parts/OCB-11.0/addons/web/controllers/main.py", line 924, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/odoo/parts/OCB-11.0/addons/web/controllers/main.py", line 916, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/odoo/parts/server-backend/base_user_role/models/role.py", line 51, in unlink
    users = self.mapped('user_id')
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/models.py", line 4408, in mapped
    recs = recs._mapped_func(operator.itemgetter(name))
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/models.py", line 4387, in _mapped_func
    vals = [func(rec) for rec in self]
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/models.py", line 4387, in <listcomp>
    vals = [func(rec) for rec in self]
  File "/home/odoo/odoo/parts/OCB-11.0/odoo/models.py", line 4654, in __getitem__
    return self._fields[key].__get__(self, type(self))
KeyError: 'user_id'
```

But it is not your fault, this bug also exists on the `8.0` branch I think (check my comment here https://github.com/OCA/server-tools/pull/922#issuecomment-346018441)
I'll make a PR to fix the branch 8.0.

**Comment by sebalix** [comment]
@novawish You can cherry-pick commits from https://github.com/OCA/server-tools/pull/1066

**Comment by novawish** [comment]
@sebalix Done, i cherry-picked and tested locally. I can not check why `ci/runbot` failed, do you have any ideas? Thank you by the way.

**Comment by sebalix** [comment]
@novawish I don't know, but I tested your PR locally and all is working well :+1:

**Review by sebalix — APPROVED** [review]

File: base_user_role/data/ir_cron.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <field name='name'>Update user roles</field>
+    <field name='interval_number'>3</field>
+    <field name='interval_type'>hours</field>
+    <field name="numbercall">-1</field>
+    <field name="active">True</field>
+    <field name="doall" eval="False"/>
+    <field name="model_id" ref="base_user_role.model_res_users_role"/>
+    <field name="state">code</field>
+    <field name="code">model.cron_update_users()</field>
+    <field name="args">()</field>
```
Comment (lasley) [inline base_user_role/data/ir_cron.xml]: Runbot is failing due to this:

> 2017-11-23 17:23:24,700 151 WARNING openerp_test odoo.models: ir.cron.create() includes unknown fields: args

https://runbot2.odoo-community.org/runbot/static/build/3311019-1-0b09c6/logs/job_20_test_all.txt

Comment (novawish) [reply base_user_role/data/ir_cron.xml]: You're right, i fixed. Thank you very much :+1:

Comment (lasley) [reply base_user_role/data/ir_cron.xml]: No prob! Admittedly I haven't reviewed beyond this, I just figured I'd help out where I could.

Looks like we're good for merge though, so I'll hit the button

Comment (novawish) [reply base_user_role/data/ir_cron.xml]: So if there is nothing else, can you approve the PR?

Comment (lasley) [reply base_user_role/data/ir_cron.xml]: My bad, I forgot to hit the button 😑

**Pull request merged**